### ISSUE TITLE
Allow retrieving suppressed bib records

### ISSUE THREAD

**Issue by escowles** [pulibrary/marc_liberation#559]

**Comment by escowles** [comment]
Blocked by https://github.com/pulibrary/voyager_helpers/issues/103

**Comment by doncat** [comment]
Including "MARC Liberation" I hope.

**Comment by carolyncole** [comment]
Here is an example of a suppressed record: https://bibdata.princeton.edu/bibliographic/10926564/items

**Comment by mzelesky** [comment]
I think we only allow retrieval of suppressed bibs through the barcode endpoint.

**Comment by kevinreiss** [comment]
That is all we need, correct? I can't think of another use case where this would helpful.

**Comment by christinach** [comment]
@kevinreiss this is a very old issue when we were still using voyager_helpers with voyager. 
As @mzelesky posted we allow retrieval of suppressed bibs through the barcode endpoint only. 
I think we can close this issue there have been some discussion in the past see https://github.com/pulibrary/bibdata/issues/710#issuecomment-670607954 and https://pulibrary.slack.com/archives/CA891BNR0/p1600875040001900.

**Comment by escowles** [comment]
The main use case that motivated this ticket is Figgy's remote metadata feature. We have had cases where content was digitized but could not be linked to the Bib records because they were suppressed, which made the workflow very convoluted and involved having to put the Bib ID into the title field and then move it to the source metadata identifier field later once the records were unsuppressed.

I think we plan to make Figgy use Orangelight instead of Bibdata to retrieve metadata at some point (@tpendragon is this right?). But I'd like to keep this ticket until we've done that or found another solution.

**Comment by tpendragon** [comment]
That's right and it's mostly there. Presumably getting suppressed records from the orangelight endpoint would be much harder.

**Comment by kevinreiss** [comment]
Yes, suppressed records would not be present in Orangelight at all. You'd need to go directly bibdata to get the metadata. We'd need to make an endpoint that would respond with suppressed records by MMS ID lookup. How critical is this feature? Could figgy default to trying OL and if the MMS ID lookup failed go to another endpoint that is able to provide suppressed record data?

**Comment by escowles** [comment]
The functionality to retrieve MARC directly from Alma may be removed, and replaced with retrieving the indexed version from Orangelight. However, it may still be possible to retrieve metadata for suppressed records from Orangelight. Pulfalight has a similiar use case, and some options are being considered: falling back on retrieving from ASpace, or indexing a minimal version of the record (e.g., just the title). We might consider similar approaches in Orangelight.

**Comment by maxkadel** [comment]
@escowles  - what I was talking about at the meeting was specifically jsonld, not MARC records - so https://bibdata.princeton.edu/bibliographic/99108564513506421/jsonld would forward to https://catalog.princeton.edu/catalog/99108564513506421.jsonld, which uses the Solr record, rather than pulling directly from Alma.

**Comment by christinach** [comment]
In the Orangelight meeting on 9/22/2022 we discussed this issue and decided to talk to DLS on what would help them and be consistent with suppressed item workflows in Pulfalight. https://docs.google.com/document/d/1GxpYrcRc6w3kMFU1XU6CEpnumFzvxFrj2GkCq_XvoWw/edit#heading=h.2n9tc3jqfuku

**Comment by kevinreiss** [comment]
Reached out to DLS today.

**Comment by kevinreiss** [comment]
Discussed at 1/19/2023 BL-OL Checkin https://docs.google.com/document/d/1GxpYrcRc6w3kMFU1XU6CEpnumFzvxFrj2GkCq_XvoWw/edit#heading=h.swvat4umzzm9.
